packages/twenty-server/src/engine/core-modules/duplicate/duplicate.service.ts

```diff
@@ -0,0 +1,159 @@
+import { Injectable } from '@nestjs/common';
+
+import { ObjectMetadataInterface } from 'src/engine/metadata-modules/field-metadata/interfaces/object-metadata.interface';
+
+import { settings } from 'src/engine/constants/settings';
+import { DUPLICATE_CRITERIA_COLLECTION } from 'src/engine/core-modules/duplicate/constants/duplicate-criteria.constants';
+import { computeObjectTargetTable } from 'src/engine/utils/compute-object-target-table.util';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+
+@Injectable()
+export class DuplicateService {
```

Comment:
nit: DuplicateRecordService?

Reply:
I don't like duplicateService much either but not a fan of DuplicateRecord which makes it sound like it's a record. I tried to find other options like DuplicateDetector, DuplicateFinder, DuplicateRecordFinder etc but not a big fan either...

Reply:
Alright! Not a big change if needed later 👍